src/lib.rs

```diff
@@ -1,66 +1,248 @@
-#![allow(non_snake_case)]
-
 //!
 //! Parsing of ABNF Core Rules
 //!
-//! See https://tools.ietf.org/html/rfc5234#appendix-B.1
+//! See <https://tools.ietf.org/html/rfc5234#appendix-B.1>
 //!
 
 pub mod complete;
 pub mod streaming;
 
-pub fn is_alpha(c: char) -> bool {
-    c.is_ascii_alphabetic()
+use nom::AsChar;
+
+/// A-Z / a-z
+///
+/// ALPHA = %x41-5A / %x61-7A
+pub fn is_alpha(c: impl AsChar) -> bool {
+    matches!(c.as_char(), '\x41'..='\x5A' | '\x61'..='\x7A')
+}
+
+#[test]
```

Comment:
In my opinion moving all tests into a module at the bottom of this file , i.e., ... 

```Rust
#[cfg(test)]
mod tests {
    // Move all tests here ...
}
```

... is more readable.

Reply:
it's was more easy to track what function was remaining to test like that. So in my opinion is the opposite it's more readable like that. Proof for complete file you added comment to remember what was missing.

Reply:
Yeah, I guess it depends what someone is looking for at a specific moment. Mostly, I am happy to "ignore" test code just by folding/hiding the test module at the bottom.

A better reason to move all tests into a module is that we want the code to be evaluated and compiled only when testing. Thus, we want to use `#[cfg(test)]`. However, interleaving the tests with the functions would result in a lot of boilerplate.

Reply:
I will move them so anyway I also need to add tarpaulin to github action that will clearly show what is missing.

Reply:
That would be awesome!